src/org/pentaho/reporting/platform/plugin/CacheManagerEndpoint.java

```diff
@@ -38,15 +37,13 @@ public Response clear() {
     try {
       final IPluginCacheManager iPluginCacheManager = PentahoSystem.get( IPluginCacheManager.class );
       final IReportContentCache cache = iPluginCacheManager.getCache();
-      cache.cleanup();
+      cache.cleanupCurrentSession();
 
       final ICacheManager cacheManager = PentahoSystem.get( ICacheManager.class );
 
       cacheManager.clearRegionCache( "report-dataset-cache" );
       cacheManager.clearRegionCache( "report-output-handlers" );
 
-      DataCacheFactory.getCache().getCacheManager().clearAll();
```

Comment:
This could be a local purge, assuming the PentahoDataCache implementation is changed to not wipe the region. 

see PentahoDataCache.PentahoDataCacheManager#clearAll and PentahoDataCacheManager#shutdown
The clearAll call there should call "killSessionCache" instead.